x/interchainquery/keeper/msg_submit_query_response_test.go

```diff
@@ -0,0 +1,165 @@
+	validMsg types.MsgSubmitQueryResponse
+	goCtx    context.Context
+	query    types.Query
+}
+
+func (s *KeeperTestSuite) SetupMsgSubmitQueryResponse() MsgSubmitQueryResponseTestCase {
+
+	// set up IBC
+	hostChainId := "GAIA"
+	s.CreateTransferChannel(hostChainId)
+
+	// define the query
+	goCtx := sdk.WrapSDKContext(s.Ctx())
+	height := int64(s.HostCtx().BlockHeight()) // start at the current height, we modify this on a per-test basis at times
+	result := []byte{0x01, 0x02, 0x03}
+	proofOps := crypto.ProofOps{}
+	fromAddress := s.TestAccs[0].String()
+	expectedId := "9792c1d779a3846a8de7ae82f31a74d308b279a521fa9e0d5c4f08917117bf3e"
+
+	_, addr, _ := bech32.DecodeAndConvert(s.TestAccs[0].String())
+	data := banktypes.CreateAccountBalancesPrefix(addr)
+	// save the query to Stride state, so it can be retrieved in the response
+	query := types.Query{
+		Id:           expectedId,
+		ConnectionId: s.TransferPath.EndpointA.ConnectionID,
+		ChainId:      hostChainId,
+		QueryType:    types.BANK_STORE_QUERY_WITH_PROOF,
+		Request:      append(data, []byte(hostChainId)...),
+		Period:       sdk.NewInt(0),
+		LastHeight:   sdk.NewInt(height),
+		CallbackId:   "withdrawalbalance",
+		// Ttl:          uint64(2545444772), // August, 2050,
+		Ttl:    uint64(1), // August, 2050,
+		Height: height,
+	}
+
+	return MsgSubmitQueryResponseTestCase{
+		validMsg: types.MsgSubmitQueryResponse{
+			ChainId:     hostChainId,
+			QueryId:     expectedId,
+			Result:      result,
+			ProofOps:    &proofOps,
+			Height:      height,
+			FromAddress: fromAddress,
+		},
+		goCtx: goCtx,
+		query: query,
+	}
+}
+
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_Successful() {
+	tc := s.SetupMsgSubmitQueryResponse()
+
+	s.App.InterchainqueryKeeper.SetQuery(s.Ctx(), tc.query)
+
+	resp, err := s.GetMsgServer().SubmitQueryResponse(tc.goCtx, &tc.validMsg)
+	s.Require().NoError(err)
+	s.Require().NotNil(resp)
+	s.Require().Equal(&types.MsgSubmitQueryResponseResponse{}, resp)
+
+	// check that the query is not in the store anymore, as it should be deleted
+	_, found := s.App.InterchainqueryKeeper.GetQuery(s.Ctx(), tc.query.Id)
+	s.Require().False(found)
+}
+
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_UnknownId() {
+	tc := s.SetupMsgSubmitQueryResponse()
+
+	tc.query.Id += "INVALID_SUFFIX" // create an invalid query id
+	s.App.InterchainqueryKeeper.SetQuery(s.Ctx(), tc.query)
+
+	resp, err := s.GetMsgServer().SubmitQueryResponse(tc.goCtx, &tc.validMsg)
+	s.Require().NoError(err)
+	s.Require().NotNil(resp)
+	s.Require().Equal(&types.MsgSubmitQueryResponseResponse{}, resp)
+
+	// check that the query is STILL in the store, as it should NOT be deleted because the query was not found
+	_, found := s.App.InterchainqueryKeeper.GetQuery(s.Ctx(), tc.query.Id)
+	s.Require().True(found)
+}
+
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_ExceededTtl() {
+	tc := s.SetupMsgSubmitQueryResponse()
+
+	tc.query.Ttl = uint64(1) // set ttl to be expired
```

Comment:
I think it's already set to 1 in setup. Although that might've been an accident?